src/lib/telemetry/client.ts

```diff
@@ -0,0 +1,144 @@
+import Queue from 'queue'
+
+export class TelemetryQueue extends Queue {}
+
+export interface TelemetryClientOptions {
+  queue: TelemetryQueue
+  endpoint?: string
+  debug?: boolean
+}
+
+export class TelemetryClient {
```

Comment:
probably makes sense to add a comment since I think this is implementing an interface for segment or something?